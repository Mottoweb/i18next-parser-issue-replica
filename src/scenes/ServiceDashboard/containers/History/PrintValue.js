import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

const PrintValue = ({ value, trigger }) => {
  const [t] = useTranslation(['translation', 'common', 'serviceDashboard']);
  if (value === 'null') {
    return (
      <span>&ndash;</span>
    );
  }
  if (['ownerContact', 'classifiedsSalesContact'].includes(trigger)) {
    return (
      <span>{value.replace(/\s\(.+\)$/, '')}</span>
    );
  }
  if (/^\[.+\]$/.test(value)) {
    return (
      <span>{value.replace(/[[\]]/g, '').split(', ').map((item) => t(item)).join(', ')}</span>
    );
  }
  return <span>{value}</span>;
};

PrintValue.propTypes = {
  value: PropTypes.string.isRequired,
  trigger: PropTypes.string,
};

PrintValue.defaultProps = {
  trigger: null,
};

export default PrintValue;
