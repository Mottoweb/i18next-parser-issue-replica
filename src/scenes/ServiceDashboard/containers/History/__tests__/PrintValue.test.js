import React from 'react';
import renderer from 'react-test-renderer';
import PrintValue from '../PrintValue';

describe('PrintValue component', () => {
  it('should print dash', () => {
    const tree = renderer.create(<PrintValue value="null" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should print ownerContact value', () => {
    const tree = renderer.create(<PrintValue value="Janine Bieri (ID 3547)" trigger="ownerContact" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should print classifiedsSalesContact value', () => {
    const tree = renderer
      .create(<PrintValue value="Matthias Mahler (ID 4527)" trigger="classifiedsSalesContact" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should print sequence', () => {
    const tree = renderer.create(<PrintValue value="[ADVERTISER, AGENCY, PUBLISHER]" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should print value', () => {
    const tree = renderer.create(<PrintValue value="Foo bar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
