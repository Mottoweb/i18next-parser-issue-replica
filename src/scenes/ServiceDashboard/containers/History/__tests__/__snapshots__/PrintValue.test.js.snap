// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintValue component should print classifiedsSalesContact value 1`] = `
<span>
  Matthias Mahler
</span>
`;

exports[`PrintValue component should print dash 1`] = `
<span>
  â€“
</span>
`;

exports[`PrintValue component should print ownerContact value 1`] = `
<span>
  Janine Bieri
</span>
`;

exports[`PrintValue component should print sequence 1`] = `
<span>
  ADVERTISER, AGENCY, PUBLISHER
</span>
`;

exports[`PrintValue component should print value 1`] = `
<span>
  Foo bar
</span>
`;
